src/instruments/qml/MuseScore/Instruments/internal/InstrumentsTreeItemDelegate.qml

```diff
@@ -245,7 +245,12 @@ Item {
                     return
                 }
 
-                model.itemRole.isVisible = !model.itemRole.isVisible
+                if (!root.isSelected) {
+                    model.itemRole.isVisible = !model.itemRole.isVisible
+                }
+                else {
```

Comment:
Please do not wrap else on a new line
https://github.com/musescore/MuseScore/wiki/CodeGuidelines#braces